src/components/radio/radio.ts

```diff
@@ -128,9 +132,16 @@ export default class IgcRadioComponent extends EventEmitterMixin<
   @watch('checked', { waitUntilFirstUpdate: true })
   protected handleChange() {
     if (this.checked) {
+      if (this.getSiblings().some((r) => r.required)) {
```

Comment:
Isn't the setRequired() in the radio group enough? Do we need this additional code here?